Tests/xSharePoint/xSharePoint.xSPSubscriptionSettings.Tests.ps1

```diff
@@ -0,0 +1,122 @@
+
+            Mock Get-SPServiceApplication { return @(@{
+                TypeName = "Some other service app type"
+            }) }
+
+            It "returns null from the Get method" {
+                Get-TargetResource @testParams | Should BeNullOrEmpty
+                Assert-MockCalled Get-SPServiceApplication -ParameterFilter { $Name -eq $testParams.Name } 
+            }
+
+        }
+
+        Context "When a service application exists and is configured correctly" {
+            Mock Get-SPServiceApplication { 
+                return @(@{
+                    TypeName = "Microsoft SharePoint Foundation Subscription Settings Service Application"
+                    DisplayName = $testParams.Name
+                    ApplicationPool = @{ Name = $testParams.ApplicationPool }
+                    Database = @{
+                        Name = $testParams.DatabaseName
+                        Server = @{ Name = $testParams.DatabaseServer }
+                    }
+                })
+            }
+
+            It "returns values from the get method" {
+                Get-TargetResource @testParams | Should Not BeNullOrEmpty
+                Assert-MockCalled Get-SPServiceApplication -ParameterFilter { $Name -eq $testParams.Name } 
+            }
+
+            It "returns true when the Test method is called" {
+                Test-TargetResource @testParams | Should Be $true
+            }
+        }
+
+        Context "When a service application exists and the app pool is not configured correctly" {
+            Mock Get-SPServiceApplication { 
+                $service = @(@{
+                    TypeName = "Microsoft SharePoint Foundation Subscription Settings Service Application"
+                    DisplayName = $testParams.Name
+                    ApplicationPool = @{ Name = "Wrong App Pool Name" }
+                    Database = @{
+                        Name = $testParams.DatabaseName
+                        Server = @{ Name = $testParams.DatabaseServer }
+                    }
+                })
+                    
+                $service = $service | Add-Member ScriptMethod Update {
+                    $Global:xSPSubscriptionServiceUpdateCalled = $true
+                } -PassThru 
+                return $service
+
+
+            }
+            Mock Get-SPServiceApplicationPool { return @{ Name = $testParams.ApplicationPool } }
+            Mock Set-SPSubscriptionSettingsServiceApplication { }
+
+            It "returns false when the Test method is called" {
+                Test-TargetResource @testParams | Should Be $false
+            }
+
+            It "calls the update service app cmdlet from the set method" {
+                Set-TargetResource @testParams
+
+                Assert-MockCalled Get-SPServiceApplicationPool
+                $Global:xSPSubscriptionServiceUpdateCalled | Should Be $true
```

Comment:
Same feedback as the other test file with setting this to be false and adding a context for no DB params